src/components/Sidebars/Payback.js

```diff
@@ -207,8 +233,13 @@ function ProxyAndAllowanceCheck({
   );
 
   useEffect(() => {
-    checkForProxyAndAllowance(maker, updateState, setHasAllowance);
-  }, [maker, account, setHasAllowance]);
+    checkForProxyAndAllowance(
+      maker,
+      updateState,
+      setHasAllowance,
+      madeInitialCheck
+    );
+  }, [maker, account, hasAllowance]);
```

Comment:
I found that `setHasAllowance()` wasn't triggering this useEffect.

Reply:
Ideally, the info about whether a given account has a proxy and an allowance wouldn't just be stored in the Payback component's state, because it gets thrown out whenever that component is unmounted. 

Instead it could be stored higher up in the component tree, or in the app's global store. Then any component who wants to know about the proxy could check the store rather than having to make its own request. And if the allowance info is kept in the store, we wouldn't have to do the same check every time we open Payback. 

Fewer network requests = more responsive UI

(Mostly just thinking out loud here; it could be applied to this code here, but let's not delay shipping the bugfix)

Reply:
> But I'm not quite clear why you would want to check again once hasAllowance has changed. For example, this would cause the check to occur again after setHasAllowance(true) is called, but at that point you already know you have a proxy and an allowance.

We would know we have a proxy yes, but are we 100% guaranteed that awaiting the transaction to set the allowance is setting the allowance?  The extra lookup to `checkForProxyAndAllowance()` triggered on the update to `hasAllowance` would guarantee that we are reading directly from the source as opposed to inferring that it is set on the basis of a completed transaction. 

It should also be noted that the `setHasAllowance(true)` after the awaited transaction is executed before the state change of `allowanceLoading: true -> false` [here](https://github.com/makerdao/mcd-cdp-portal/blob/66c0973853ecb4802ce11bf8ffd427eeebfe3068/src/components/Sidebars/Payback.js#L197). In one respect this is done for UI purposes so the toggle (which is predicated to be active on `allowanceLoading || hasAllowance`) isn't switched as inactive while the real value for the allowance is looked up.

I do agree with refactoring so that the state is stored further up the component tree. I have mentioned it in this mornings work breakdown call as a task we can put on the backlog.

Reply:
> The extra lookup to `checkForProxyAndAllowance()` triggered on the update to `hasAllowance` would guarantee that we are reading directly from the source as opposed to inferring that it is set on the basis of a completed transaction....

I think the only reason we wouldn't trust it to be set after the tx is mined is Infura node lag issues. We can mitigate that with
```js 
await txMgr.confirm(txPromise, n) // n is some number of blocks
```
This way, the workaround for Infura issues is more contained, and other code can simply trust that when `hasAllowance` is true, it's really true.

Reply:
I think if you do this, then the rest of the fix is just the `setAllowance was never passed`, and the `madeInitialCheck` stuff isn't needed?

Reply:
Yes, that's all. I'll make that fix